crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -577,15 +610,29 @@ impl WritebackCache {
         epoch: EpochId,
         digest: TransactionDigest,
     ) -> SuiResult {
-        let Some((_, outputs)) = self.dirty.pending_transaction_writes.remove(&digest) else {
+        let DashMapEntry::Occupied(occupied) = self.dirty.pending_transaction_writes.entry(digest)
```

Comment:
Does this hold a lock to the entire `pending_transaction_writes` until it's removed? Would it be possible to clone it out to avoid holding the lock during db writes?

Reply:
It only holds a lock to one shard of the map. The db write could still get very slow if rocksdb starts throttling writes, but the resulting back pressure on writers may actually be a good thing in that case. Anyway, i'm going to re-work this code a bit to handle per-checkpoint batch writes, so we can think harder about this at that point